### ISSUE TITLE
add an action for fast-forward merge

### ISSUE THREAD

**Issue by lsf37** [seL4/ci-actions#127]
Or possibly a github-app, but it should be possible to do in a `pull_request_target` action. Might just be a bit slow, and show up in the status checks even when not running.

Requirements:
- only people with write access to the repo should be able to trigger an actual merge
- abort when there is no clean ff merge

Why?
- preserves gpg-signed commits and commit hashes

**Comment by kent-mcleod** [comment]
Did GitHub used to support fast-forward merging and then remove it?

**Comment by lsf37** [comment]
There used to be a GitHub app at least that supported it, but it is no longer maintained. I don't remember if GitHub itself had a ff-merge button, but I don't think so.

**Comment by kent-mcleod** [comment]
Right.  I guess we just simulated it with ff-merge on bitbucket which then pushed to github.  And I guess GitHub doesn't allow it because they want to record who the committer was.

**Comment by lsf37** [comment]
Just logging an observation here: manual fast-forward merge on the command line still works fine, even with PR review and test requirements:

```
git checkout master
git pull
git merge --ff-only <my-pr-branch>
git push
```

does work fine if you have sufficient rights to merge the pull request and if the required conditions for merging the PR are achieved. This makes it possible for instance to use gpg-signed commits (without GitHub invalidating the signature on rebase merges).

It'd still be nice if there was an UI thing that did that for us.
